formatters/swiftcov/parse.go

```diff
@@ -0,0 +1,128 @@
+package swiftcov
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/codeclimate/test-reporter/env"
+	"github.com/codeclimate/test-reporter/formatters"
+	"github.com/markbates/pop/nulls"
+	"github.com/pkg/errors"
+)
+
+// Formatter collects SwiftCov files, parses them, then formats them into a single report.
+type Formatter struct {
+	FileNames   []string
+	SourceFiles []formatters.SourceFile
+}
+
+var searchPaths = []string{"./"}
+var search = ".swift.gcov" // look for these file extensions
+
+// Search searches the designated paths for SwiftCov files,
+// appending them to the list of filenames.
+func (f *Formatter) Search(paths ...string) (string, error) {
```

Comment:
I'm open to extracting the "find files by extension" functionality out, but it's not clear to me where common code like should live.